spec/log_search_spec.rb

```diff
@@ -1,18 +1,18 @@
 require 'spec_helper'
 
-describe LogSearch do
+shared_examples :search_engine do
```

Comment:
Since Ruby is duck-typed, it's generally considered an anti-pattern to create a base class that represents and interface or abstract class (a la Java, C#, or C++). It's impossible to enforce base classes of that nature. One common practice is to used shared test groups. The tests define the behavior that would otherwise be expected of the abstract class or interface. The shared tests are then used to test all implementations. In RSpec we use `shared_examples` to accomplish this.